src/bluesearch/entrypoint/database/add.py

```diff
@@ -59,9 +77,22 @@ def run(
         # This branch never reached because of `choices` in `argparse`
         raise ValueError(f"Unrecognized database type {db_type}.")  # pragma: nocover
 
-    with open(path, "rb") as f:
-        article = pickle.load(f)  # nosec
+    inputs: Iterable[Path]
+    if parsed_path.is_file():
+        inputs = [parsed_path]
+    elif parsed_path.is_dir():
+        inputs = parsed_path.glob("*")
```

Comment:
I do like your idea, but I thought that this would set a precedent for the similar `glob("*")` in `parse.py`.

And in `parse.py` it is much harder to guess the extension. 
- First, because the extension depends on the format — e.g. `*.json` for `cord19-json` and `*.xml` for `pmc-xml`.
- Second, because even for the same format you may have different extensions — e.g. `*.xml` and `*.nxml` are both used for `pmc-xml` files).

Reply:
I see the problme for `parse`, but here we'll always have one single fixed extension, right?

Regarding `parse`, there we can actually run into the same trouble of having random foreign files being around and crashing our parsers. I don't see a super straightforward solution for `parse` except catching potential exceptions... Could be a separate issue.

Reply:
Done in 10e2e28.
In addition to that, I created a dedicated issue https://github.com/BlueBrain/Search/issues/445 for `bbs_database parse` and I tried to also provide some possible solutions.

Reply:
awesome, thanks!